File: 20230828/java/mschoi.java
import java.io.*;
import java.util.StringTokenizer;

class Solution {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int T = Integer.parseInt(st.nextToken());
        for (int test_case = 1; test_case <= T; test_case++) {
            st = new StringTokenizer(br.readLine());
            int N = Integer.parseInt(st.nextToken());
            int V = Integer.parseInt(st.nextToken());

            int answer;
            if (V == 1) {
                answer = (int) (Math.log(N) / Math.log(2));
                if (N == 1) {
                    answer = 0;
                }
            } else {
                answer = (int) (Math.log(V) / Math.log(2));
                int count = 0;
                while (V > 3) { // 루트 기준 왼쪽 서브트리인지 오른쪽 서브트리인지 판단
                    V /= 2;
                }
                if (V == 2) { // 왼쪽에서 올라왔다
                    if (N >= 3) {
                        answer++;
                        answer += (int) (Math.log(N / 3) / Math.log(2));
                    }
                } else { // 오른쪽에서 올라왔다
                    answer += (int) (Math.log(N) / Math.log(2));
                }
            }
            bw.write("#" + test_case + " " + answer + "\n");
        }

        bw.flush();
        br.close();
        bw.close();
    }
}

// 66,368 kb
// 267 ms

File: 20230828/python/mschoi.py
import math

T = int(input())
for test_case in range(1, T + 1):
    N, V = map(int, input().split())

    # 루트 노드라면 가장 최대 깊이 반환
    if V == 1:
        answer = int(math.log2(N))
        if N == 1:
            answer = 0
    else: # 루트 노드가 아니라면
        # 루트 노드까지 올라가서 최대한 깊이 내려가기
        answer = math.floor(math.log2(V))
        if V % 2 == 0: # 짝수라면 왼쪽에서 올라왔다
            if N >= 3:
                answer += 1
                answer += math.floor(math.log2(N/3))
        else: # 홀수라면 오른쪽에서 올라왔다
            answer += math.floor(math.log2(N))
    print("#" + str(test_case) + " " + str(answer))

# 시간초과
처음 테케 : 실행 시간 0.14292s
